articles/build-a-ticketing-app-with-adonisjs-and-vuejs/index.md

```diff
@@ -0,0 +1,902 @@
+
+### Prerequisites
+
+Before continuing with this article, you should have previous experiences with the following:
+
+1. Basic understanding of TypeScript and Node.js
+2. Basic [understanding of AdonisJS](https://masteringbackend.com/posts/adonisjs-tutorial-the-ultimate-guide)
+3. Basic understanding of VueJS
+
+### Goals of the Tutorial
+
+I will be showing you how to build a Ticketing App with AdonisJS 5 and VueJS. I will build a Ticketing Website that users can find events around you, register for a free event, buy tickets, sell tickets online and promote their event worldwide.
+
+I will also cover best practices in structuring and building your frontend with VueJS when using AdonisJS as your backend.
+
+Students will learn how to build a real-world app with AdonisJS and VueJS
+Students will learn how to handle authentication and authorization with AdonisJS and VueJS
+Students will learn how to integrate AdonisJS with VueJS.
+Students will learn how to build and structure a Ticketing System in AdonisJS and VueJS
+Students will learn best practices in building performance-oriented Web apps with AdonisJS.
+
+### Step 1 — Setting up Adonis.js 5
+
+Setting up AdonisJS 5 is very easy as it requires a simple NPM command below, if you want to learn more about installing and setting up AdonisJS 5, check out [Building a RESTful API with Adonis.js](https://www.section.io/engineering-education/build-a-restful-api-with-adonisjs/)
+
+```bash
+npm init adonis-ts-app adonisjs-ticketing-system-api
+```
+
+After creating the AdonisJS application, move inside the project directory and install the following packages to set up the API.
+
+```bash
+cd adonisjs-ticketing-system-api
+```
+
+Now, we need to set up our database to connect with the new Adonisjs application.
+
+#### Creating the database
+
+Our ticketing system application needs a database for storing, retrieving, updating and deleting data.
+
+To install the database, we need to first create the database by using any of these [database clients](https://masteringbackend.com/posts/top-10-database-clients-for-developers) and install the `Lucid` ORM.
+
+```bash
+npm i @adonisjs/lucid@alpha
+```
+
+After installing the ORM, set it up by running the following command.
+
+```bash
+node ace invoke @adonisjs/lucid
+```
+
+After running the above command, follow the instructions and update your `.env` file.
+
+```bash
+DB_CONNECTION=mysql
+MYSQL_USER= //DB_USER
+MYSQL_HOST=localhost
+MYSQL_DB_NAME= //DB_NAME
+MYSQL_PORT=3306
+MYSQL_PASSWORD= //DB_PASSWORD
+```
+
+If you encounter this error `Client does not support the authentication protocol requested by the server;` while testing your API, follow these steps to solve it.
+
+```bash
+npm install mysql2
+```
+
+To set up authentication in Adonisjs is done by installing a package and setting it up, you can read through how to set up authentication [here](https://www.section.io/engineering-education/build-a-restful-api-with-adonisjs/).
```

Comment:
When you are referencing some other Section article, you need not type out the entire URL. Instead, you could use:

`/engineering-education/build-a-restful-api-with-adonisjs/`

Reply:
Done